SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -0,0 +1,530 @@
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_overflow_multicast_rpcs_to_the_service_THEN_rpc_push_result_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH);
+
+	// Send RPCs to the point where we will overflow
+	int RPCsToSend = static_cast<int>(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::NetMulticast));
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	}
+
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	TestTrue("Push RPC returned expected results", (Result == SpatialGDK::EPushRPCResult::Success));
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_server_unreliable_rpcs_to_the_service_THEN_payloads_are_writen_correctly_to_component_updates)
+{
+	TArray<EntityPayload> EntityPayloads;
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_1, SimplePayload));
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_2, SimplePayload));
+
+	TArray<Worker_EntityId> EntityIdArray;
+	EntityIdArray.Add(RPCTestEntityId_1);
+	EntityIdArray.Add(RPCTestEntityId_2);
+
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService(EntityIdArray, CLIENT_AUTH);
+	for (const EntityPayload& EntityPayloadItem : EntityPayloads)
+	{
+		RPCService.PushRPC(EntityPayloadItem.EntityId, ERPCType::ServerUnreliable, SimplePayload);
+	}
+
+	TArray<SpatialGDK::SpatialRPCService::UpdateToSend> UpdateToSendArray = RPCService.GetRPCsAndAcksToSend();
+
+	bool bTestPassed = true;
+	if (UpdateToSendArray.Num() != EntityPayloads.Num())
+	{
+		bTestPassed = false;
+	}
+	else
+	{
+		for (int i = 0; i < EntityPayloads.Num(); ++i)
+		{
+			if (!CompareUpdateToSendAndEntityPayload(UpdateToSendArray[i], EntityPayloads[i], ERPCType::ServerUnreliable, 1))
+			{
+				bTestPassed = false;
+				break;
+			}
+		}
+	}
+
+	TestTrue("UpdateToSend have expected payloads", bTestPassed);
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_no_authority_over_rpc_endpoint_WHEN_push_client_reliable_rpcs_to_the_service_THEN_component_data_matches_payload)
+{
+	// Create RPCService with empty component view
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({}, INVALID);
+
+	RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientReliable, SimplePayload);
+
+	Worker_ComponentData ComponentData = GetComponentDataOnEntityCreationFromRPCService(RPCService, RPCTestEntityId_1, ERPCType::ClientReliable);
+	bool bTestPassed = CompareComponentDataAndEntityPayload(ComponentData, EntityPayload(RPCTestEntityId_1, SimplePayload), ERPCType::ClientReliable, 1);
+	TestTrue("Entity creation test returned expected results", bTestPassed);
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_no_authority_over_rpc_endpoint_WHEN_push_multicast_rpcs_to_the_service_THEN_initially_preset_set)
```

Comment:
`initially_present` or `initial_preset`?

Reply:
initially_present is the name of the property on the component I think so it probably is the correct thing to use. It just sounds a little weird if you don't have full context.